### ISSUE TITLE
Jax-based BFGS minimize fails to converge, head-to-head with SciPy

### ISSUE THREAD

**Issue by gully** [google/jax#4594]
Hello,
Thank you for making Jax!
I recently compared the jax-based BFGS minimize with the scipy-based BFGS minimize on an astronomical scene modeling demo of 3 parameters.  I find that the Jax version failed with error code 3 "zoom failed", while the SciPy version succeeds:

https://gist.github.com/gully/577c6d8eb96d26379f8e6ce89b9d2a2e 

This might be a corner case, since the inverse-Hessian appears to be non-Positive-semidefinite, perhaps pointing towards some other problem.  At first I thought it was a `float32` issue, but I'm pretty sure all my arrays are `float64` (I change the config at the top).  I'm using jax version 2.3, and jaxlib 0.1.55 with CUDA 10.2.  

Any guidance is appreciated!  Thanks again.

**Comment by Joshuaalbert** [comment]
Hi @gully glad to have you trying it out. It looks like the jacobian is very large, `[    8428.3213677 , -2903921.86042314,  6528682.31072694]` in magnitude at the initial point. The output says number of iterations is 1. Meaning that it failed to do a line search on the first iteration because the line search didn't let small enough values be tried. I'll look into adjusting this behaviour to favour very steep scapes better. But for now just normalise your loss function by some suitably large number like `jnp.linalg.norm(jnp.array([    8428.3213677 , -2903921.86042314,  6528682.31072694]))`. This will let smaller step sizes work at the start.

**Comment by gully** [comment]
@Joshuaalbert Thank you so much for a careful response and looking through the notebook.  I was following your PR to get this functionality merged.  I will try as you describe.

**Comment by Joshuaalbert** [comment]
Glad this solved it. To address the underlying issue of line search in regions of large gradients I'll need to adjust the linesearch. In principle once the Hessian inverse starts to approximate the actual Hessian inverse (typically several iterations in) the step size tends to unity. But, at the start the optimal step size can be very large or very small which can be a problem. Maybe simply letting a large range of step sizes during zoom part of line search would solve this.

**Comment by Joshuaalbert** [comment]
Can be closed
